### DISCUSSION TITLE
Github copilot extension button

### DISCUSSION THREAD

**Discussion by ghost** [VsixCommunity/Community.VisualStudio.Toolkit#504] [Q&A] [answered]
Hi, i'm very curious on how to create a button like the Copilot extension one;

1. How is that part of the IDE called (top-right toolbar)? 
2. How can a button be that large, having an icon and text at the same time? Perhaps its a large .png? I can only get to show icons with the same size than the save, save all, undo default buttons...
3. How to add a dropdown functionality when on-click? Same that with a normal button? (menus, groups etc.)
  

![imagen](https://github.com/VsixCommunity/Community.VisualStudio.Toolkit/assets/137062705/546fa272-22f3-4c81-9908-14ae23df3e62)

**Comment by reduckted** [comment, marked as answer]
1. Copilot has the advantage of being built in to Visual Studio rather than being an extension like one that you or I can create. That means it can put buttons and icons and other things in places that an ordinary extension cannot. If you really want to put a button in that location or somewhere similar, take a look at #499 where I explained how it can be done (I wouldn't recommend doing it though)
2. Because it's not a regular toolbar button, it can be whatever size it wants. I haven't looked at what WPF elements it's using, but I imagine it would be a [`CrispImage`](https://learn.microsoft.com/en-us/dotnet/api/microsoft.visualstudio.imaging.crispimage?view=visualstudiosdk-2022).
3. For an example of how to show a context menu, take a look at this PR: https://github.com/madskristensen/DocumentMargin/pull/2

**Reply by ghost** [reply]
Thank you so much for the quick reply. Really helpful information, have a great day.
